client/ayon_sitesync/launch_hooks/pre_copy_last_published_workfile.py

```diff
@@ -60,14 +54,13 @@ def execute(self):
 
         # Get data
         project_name = self.data["project_name"]
-        asset_name = self.data["asset_name"]
         task_name = self.data["task_name"]
         task_type = self.data["task_type"]
         host_name = self.application.host_name
 
         # Check settings has enabled it
         project_settings = get_project_settings(project_name)
```

Comment:
Project settings are availalble in `self.data` and you can use function `should_use_last_workfile_on_launch` from `ayon_core.pipeline.workfile`.

Reply:
Reworked according to recommendation.